File: api/src/main/java/com/ruby/rubia_server/core/entity/Message.java
package com.ruby.rubia_server.core.entity;

import com.ruby.rubia_server.core.enums.MessageStatus;
import com.ruby.rubia_server.core.enums.SenderType;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.time.LocalDateTime;
import java.util.UUID;

import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "messages")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Message {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "conversation_id", nullable = false)
    private Conversation conversation;

    @Column(columnDefinition = "TEXT")
    private String content;

    @Enumerated(EnumType.STRING)
    @JdbcTypeCode(SqlTypes.NAMED_ENUM)
    @Column(name = "sender_type", nullable = false)
    private SenderType senderType;

    @Column(name = "sender_id")
    private UUID senderId;

    @Enumerated(EnumType.STRING)
    @JdbcTypeCode(SqlTypes.NAMED_ENUM)
    @Column(name = "status", nullable = false)
    @Builder.Default
    private MessageStatus status = MessageStatus.SENT;

    @Column(name = "delivered_at")
    private LocalDateTime deliveredAt;

    @Column(name = "read_at")
    private LocalDateTime readAt;

    @Column(name = "external_message_id")
    private String externalMessageId;

    @Column(name = "is_ai_generated")
    private Boolean isAiGenerated;

    @Column(name = "ai_confidence")
    private Double aiConfidence;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "ai_agent_id")
    private AIAgent aiAgent;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "message_template_id")
    private MessageTemplate messageTemplate;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "campaign_contact_id")
    private CampaignContact campaignContact;

    // A relação com ConversationMedia permanece como está (One-to-One opcional)
    @OneToOne(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @JoinColumn(name = "conversation_media_id", unique = true)
    private ConversationMedia media;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        if (createdAt == null) {
            createdAt = LocalDateTime.now();
        }
    }
}

File: api-dotnet/Rubia.Server/Entities/Message.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Rubia.Server.Enums;

namespace Rubia.Server.Entities;

[Table("messages")]
public class Message : BaseEntity
{
    [Column("content", TypeName = "TEXT")]
    public string? Content { get; set; }

    [Column("sender_type")]
    [Required]
    public SenderType SenderType { get; set; }

    [Column("sender_id")]
    public Guid? SenderId { get; set; }

    [Column("status")]
    [Required]
    public MessageStatus Status { get; set; } = MessageStatus.Sent;

    [Column("delivered_at")]
    public DateTime? DeliveredAt { get; set; }

    [Column("read_at")]
    public DateTime? ReadAt { get; set; }

    [Column("external_message_id")]
    public string? ExternalMessageId { get; set; }

    [Column("is_ai_generated")]
    public bool? IsAiGenerated { get; set; }

    [Column("ai_confidence")]
    public double? AiConfidence { get; set; }

    [Column("sentiment")]
    [MaxLength(20)]
    public string? Sentiment { get; set; }

    [Column("keywords")]
    public string? Keywords { get; set; }

    // Navigation properties
    [Column("conversation_id")]
    [Required]
    public Guid ConversationId { get; set; }
    
    [ForeignKey("ConversationId")]
    public virtual Conversation Conversation { get; set; } = null!;

    [Column("ai_agent_id")]
    public Guid? AiAgentId { get; set; }
    
    [ForeignKey("AiAgentId")]
    public virtual AIAgent? AiAgent { get; set; }

    [Column("message_template_id")]
    public Guid? MessageTemplateId { get; set; }
    
    [ForeignKey("MessageTemplateId")]
    public virtual MessageTemplate? MessageTemplate { get; set; }

    [Column("campaign_contact_id")]
    public Guid? CampaignContactId { get; set; }
    
    [ForeignKey("CampaignContactId")]
    public virtual CampaignContact? CampaignContact { get; set; }

    [Column("conversation_media_id")]
    public Guid? ConversationMediaId { get; set; }
    
    [ForeignKey("ConversationMediaId")]
    public virtual ConversationMedia? Media { get; set; }
}